From 668a1513d5f3f00b255dd5fb430c21f43354fdc5 Mon Sep 17 00:00:00 2001
From: Vincent Becker <vincentx.becker@intel.com>
Date: Fri, 10 Aug 2012 14:50:03 +0200
Subject: [PATCH 01/49] [PORT FROM MAIN] [VIBRA] Remove vibrator AOSP
 implementation from libhardware_legacy

BZ: 49760

To enable the vendor implementation of the vibrator HAL, it is needed
to use hardware modules, a default one for AOSP and the vendor one.
The implementation in libhardware_legacy is no more needed so remove
it from the file system.

Change-Id: Id7c809033fafecc4a3036829bf1a00072bca7826
Origin-Change-Id: I131a1911d26325a31968eff8c6ff693ee819c892
JB-integ-Change-Id: Ib285994d03a00ab646e1a16d5714553acb4222d9
Signed-off-by: Vincent Becker <vincentx.becker@intel.com>
Reviewed-on: http://android.intel.com:8080/75585
Reviewed-by: Barthes, FabienX <fabienx.barthes@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 Android.mk          |    2 +-
 vibrator/Android.mk |    4 ---
 vibrator/vibrator.c |   75 ---------------------------------------------------
 3 files changed, 1 insertions(+), 80 deletions(-)
 delete mode 100644 vibrator/Android.mk
 delete mode 100644 vibrator/vibrator.c

diff --git a/Android.mk b/Android.mk
index 1faf6e2..e2b35c6 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,7 +1,7 @@
 # Copyright 2006 The Android Open Source Project
 
 # Setting LOCAL_PATH will mess up all-subdir-makefiles, so do it beforehand.
-legacy_modules := power uevent vibrator wifi qemu qemu_tracing
+legacy_modules := power uevent wifi qemu qemu_tracing
 
 SAVE_MAKEFILES := $(call all-named-subdir-makefiles,$(legacy_modules))
 LEGACY_AUDIO_MAKEFILES := $(call all-named-subdir-makefiles,audio)
diff --git a/vibrator/Android.mk b/vibrator/Android.mk
deleted file mode 100644
index 6681f84..0000000
--- a/vibrator/Android.mk
+++ /dev/null
@@ -1,4 +0,0 @@
-# Copyright 2006 The Android Open Source Project
-
-LOCAL_SRC_FILES += vibrator/vibrator.c
-
diff --git a/vibrator/vibrator.c b/vibrator/vibrator.c
deleted file mode 100644
index f946ce1..0000000
--- a/vibrator/vibrator.c
+++ /dev/null
@@ -1,75 +0,0 @@
-/*
- * Copyright (C) 2008 The Android Open Source Project
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-#include <hardware_legacy/vibrator.h>
-#include "qemu.h"
-
-#include <stdio.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <errno.h>
-
-#define THE_DEVICE "/sys/class/timed_output/vibrator/enable"
-
-int vibrator_exists()
-{
-    int fd;
-
-#ifdef QEMU_HARDWARE
-    if (qemu_check()) {
-        return 1;
-    }
-#endif
-
-    fd = open(THE_DEVICE, O_RDWR);
-    if(fd < 0)
-        return 0;
-    close(fd);
-    return 1;
-}
-
-static int sendit(int timeout_ms)
-{
-    int nwr, ret, fd;
-    char value[20];
-
-#ifdef QEMU_HARDWARE
-    if (qemu_check()) {
-        return qemu_control_command( "vibrator:%d", timeout_ms );
-    }
-#endif
-
-    fd = open(THE_DEVICE, O_RDWR);
-    if(fd < 0)
-        return errno;
-
-    nwr = sprintf(value, "%d\n", timeout_ms);
-    ret = write(fd, value, nwr);
-
-    close(fd);
-
-    return (ret == nwr) ? 0 : -1;
-}
-
-int vibrator_on(int timeout_ms)
-{
-    /* constant on, up to maximum allowed time */
-    return sendit(timeout_ms);
-}
-
-int vibrator_off()
-{
-    return sendit(0);
-}
-- 
1.7.9

